test/test_suite.py

```diff
@@ -850,3 +851,320 @@ class StringTest(_StringTest):
+        """
+        SPANNER OVERRIDE:
+
+        Spanner throws an error 400 Value has type FLOAT64 which cannot be
+        inserted into column x, which has type NUMERIC for value 15.7563.
+        Overriding the test to remove the failure case.
+        """
+        self._do_test(
+            Numeric(precision=8, scale=4),
+            [decimal.Decimal("15.7563")],
+            [decimal.Decimal("15.7563")],
+        )
+
+    def test_numeric_as_float(self):
+        """
+        SPANNER OVERRIDE:
+
+        Spanner throws an error 400 Value has type FLOAT64 which cannot be
+        inserted into column x, which has type NUMERIC for value 15.7563.
+        Overriding the test to remove the failure case.
+        """
+
+        self._do_test(
+            Numeric(precision=8, scale=4, asdecimal=False),
+            [decimal.Decimal("15.7563")],
+            [15.7563],
+        )
+
+    @requires.floats_to_four_decimals
+    def test_float_as_decimal(self):
+        """
+        SPANNER OVERRIDE:
+
+        Cloud Spanner supports tables with an empty primary key, but
+        only a single row can be inserted into such a table -
+        following insertions will fail with `Row [] already exists".
+        Overriding the test to avoid the same failure.
+        """
+        self._do_test(
+            Float(precision=8, asdecimal=True), [15.7563], [decimal.Decimal("15.7563")],
+        )
+
+        self._do_test(
+            Float(precision=8, asdecimal=True),
+            [decimal.Decimal("15.7563")],
+            [decimal.Decimal("15.7563")],
+        )
+
+    def test_float_as_float(self):
+        """
+        SPANNER OVERRIDE:
+
+        Cloud Spanner supports tables with an empty primary key, but
+        only a single row can be inserted into such a table -
+        following insertions will fail with `Row [] already exists".
+        Overriding the test to avoid the same failure.
+        """
+        self._do_test(
+            Float(precision=8),
+            [15.7563],
+            [15.7563],
+            filter_=lambda n: n is not None and round(n, 5) or None,
+        )
+
+        self._do_test(
+            Float(precision=8),
+            [decimal.Decimal("15.7563")],
+            [15.7563],
+            filter_=lambda n: n is not None and round(n, 5) or None,
+        )
+
+    @requires.precision_numerics_general
+    def test_precision_decimal(self):
+        """
+        SPANNER OVERRIDE:
+
+        Cloud Spanner supports tables with an empty primary key, but
+        only a single row can be inserted into such a table -
+        following insertions will fail with `Row [] already exists".
+        Overriding the test to avoid the same failure.
+
+        Remove an extra digits after decimal point as cloud spanner is
+        capable of representing an exact numeric value with a precision
+        of 38 and scale of 9.
+        """
+        self._do_test(
+            Numeric(precision=18, scale=12),
```

Comment:
We are overriding a number of tests because Spanner does not support `scale>9`. But these tests still uses `Numeric` with `scale=12` or `scale=14`. I think it makes sense to change these `Numeric`s to use `scale=9`. WDYT?